iml-orm/src/sfa/mod.rs

```diff
@@ -0,0 +1,491 @@
+    }
+}
+
+impl From<i16> for HealthState {
+    fn from(x: i16) -> Self {
+        match x {
+            0 => Self::None,
+            1 => Self::Ok,
+            2 => Self::NonCritical,
+            3 => Self::Critical,
+            _ => Self::Unknown,
+        }
+    }
+}
+
+#[cfg(feature = "postgres-interop")]
+impl<DB, ST> Queryable<ST, DB> for HealthState
+where
+    DB: Backend,
+    i16: Queryable<ST, DB>,
+{
+    type Row = <i16 as Queryable<ST, DB>>::Row;
+
+    fn build(row: Self::Row) -> Self {
+        i16::build(row).into()
+    }
+}
+
+#[derive(
+    serde::Serialize, serde::Deserialize, Debug, Clone, Copy, PartialEq, Eq, Ord, PartialOrd,
+)]
+#[cfg_attr(feature = "postgres-interop", derive(AsExpression))]
+#[cfg_attr(feature = "postgres-interop", sql_type = "SmallInt")]
+pub enum JobType {
+    TypeInitialize = 0,
+    TypeRebuild = 1,
+    TypeRebuildFract = 2,
+    TypeRebuildDist = 3,
+    TypeErase = 4,
+    TypeDelete = 5,
+    TypeFailover = 6,
+    TypeMove = 7,
+    TypeMigrate = 8,
+    TypeVerify = 9,
+    TypeVerifyForce = 10,
+    TypeVerifyOnce = 11,
+    TypeVerifyNoCorrect = 12,
+    TypeNonDestructiveInitialize = 13,
+    TypeCopyback = 14,
+    TypeRebuildFast = 15,
+    TypeFailback = 16,
+    TypeRebuildMixed = 17,
+    TypeUnknown = 255,
+}
+
+#[cfg(feature = "postgres-interop")]
+impl<DB> ToSql<SmallInt, DB> for JobType
+where
+    DB: Backend,
+    i16: ToSql<SmallInt, DB>,
+{
+    fn to_sql<W: Write>(&self, out: &mut Output<W, DB>) -> serialize::Result {
+        (*self as i16).to_sql(out)
+    }
+}
+
+impl From<i16> for JobType {
+    fn from(x: i16) -> Self {
+        match x {
+            0 => Self::TypeInitialize,
+            1 => Self::TypeRebuild,
+            2 => Self::TypeRebuildFract,
+            3 => Self::TypeRebuildDist,
+            4 => Self::TypeErase,
+            5 => Self::TypeDelete,
+            6 => Self::TypeFailover,
+            7 => Self::TypeMove,
+            8 => Self::TypeMigrate,
+            9 => Self::TypeVerify,
+            10 => Self::TypeVerifyForce,
+            11 => Self::TypeVerifyOnce,
+            12 => Self::TypeVerifyNoCorrect,
+            13 => Self::TypeNonDestructiveInitialize,
+            14 => Self::TypeCopyback,
+            15 => Self::TypeRebuildFast,
+            16 => Self::TypeFailback,
+            17 => Self::TypeRebuildMixed,
+            _ => Self::TypeUnknown,
+        }
+    }
+}
+
+#[cfg(feature = "postgres-interop")]
+impl<DB, ST> Queryable<ST, DB> for JobType
+where
+    DB: Backend,
+    i16: Queryable<ST, DB>,
+{
+    type Row = <i16 as Queryable<ST, DB>>::Row;
+
+    fn build(row: Self::Row) -> Self {
+        i16::build(row).into()
+    }
+}
+#[derive(
+    serde::Serialize, serde::Deserialize, Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd,
+)]
+#[cfg_attr(feature = "postgres-interop", derive(AsExpression))]
+#[cfg_attr(feature = "postgres-interop", sql_type = "SmallInt")]
+pub enum JobState {
```

Comment:
Here and in `JobType` would it make sense to omit the prefixes (`State` and `Type`)?

Reply:
I want these to be easily referenceable in the SFA API docs. There they refer to each enum for this property with the JOB_TYPE_* prefix.

Reply:
On second thought, I'll remove it. The enum identifier will be there in most cases.